src/library/scala/collection/Iterator.scala

```diff
@@ -700,6 +700,7 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
       private def store(a: A): Unit = {
         if (lookahead == null) lookahead = new mutable.Queue[A]
         lookahead += a
+        ()
```

Comment:
This would benefit from the mitigations in nonunit warning, if it's returning a singleton you already have, you don't mind "discarding" it.

Reply:
💯 reminder to self.